File: 2090. K Radius Subarray Averages/Solution.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int n = nums.size();
        long long curr = 0;
        vector<int> res(n, -1);
        for (int i = 0; i < min(n, 2 * k); i++) curr += nums[i];
        for (int i = k; i < n - k; i++) {
            curr += nums[i + k];
            res[i] = (int) (curr / (2 * k + 1));
            curr -= nums[i - k];
        }
        return res;
    }
};

File: 2090. K Radius Subarray Averages/Solution.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        res = [-1 for i in range(n)]
        curr = sum(nums[:2 * k])
        for i in range(k, n - k):
            curr += nums[i + k]
            res[i] = curr // (2 * k + 1)
            curr -= nums[i - k]
        return res
